src/psyclone/psyir/backend/sir.py

```diff
@@ -0,0 +1,408 @@
+            raise VisitorError("Child of loop should be a single loop.")
+
+        # Check second loop has a single loop as a child.
+        loop_content = loop_content[0].loop_body.children
+        if not (len(loop_content) == 1 and
+                isinstance(loop_content[0], NemoLoop)):
+            raise VisitorError(
+                "Child of child of loop should be a single loop.")
+
+        # Check third loop has a single NemoKern as a child.
+        loop_content = loop_content[0].loop_body.children
+        if not (len(loop_content) == 1 and
+                isinstance(loop_content[0], NemoKern)):
+            raise VisitorError(
+                "Child of child of child of loop should be a NemoKern.")
+
+        # The interval values are hardcoded for the moment (see #470).
+        result = ("{0}interval = makeInterval(Interval.Start, Interval.End, "
+                  "0, 0)\n".format(self._nindent))
+        result += ("{0}bodyAST = makeAST([\n".format(self._nindent))
+        self._depth += 1
+        result += self.nemokern_node(loop_content[0])
+        self._depth -= 1
+        if result[-1] == "\n" and result[-2] == ",":
+            # Remove the additional comma as this is the last entry in
+            # makeAST.
+            result = result[:-2] + "\n"
+        result += "{0}])\n".format(self._nindent)
+        # For the moment there is a hard coded assumption that the
+        # vertical looping is in the forward (1..n) direction (see
+        # #470).
+        result += ("{0}verticalRegionFns.append(makeVerticalRegionDeclStmt("
+                   "bodyAST, interval, VerticalRegion.Forward))\n"
+                   "".format(self._nindent))
+        return result
+
+    def nemokern_node(self, node):
+        '''NEMO kernels are a group of nodes collected into a schedule
+        so simply call the nodes in the schedule.
+
+        :param node: a NemoKern PSyIR node.
+        :type node: :py:class:`psyclone.nemo.NemoKern`
+
+        :returns: the SIR code as a string.
+        :rtype: str
+
+        '''
+        result = ""
+        schedule = node.get_kernel_schedule()
+        for child in schedule.children:
+            result += self._visit(child)
+        return result
+
+    def nemoinvokeschedule_node(self, node):
+        '''This method is called when a NemoInvokeSchedule instance is found
+        in the PSyIR tree.
+
+        :param node: a KernelSchedule PSyIR node.
+        :type node: :py:class:`psyclone.psyGen.KernelSchedule`
+
+        :returns: the SIR code as a string.
+        :rtype: str
+
+        '''
+        result = "# PSyclone autogenerated SIR Python\n"
+        result += "verticalRegionFns = []\n"
+        # The stencil name is currently hardcoded.
+        result += "stencilname = \"psyclone\"\n"
+
+        exec_statements = ""
+        for child in node.children:
+            exec_statements += self._visit(child)
+        result += (
+            "{0}\n"
```

Comment:
Strange line break. Why not just `"{0}\n".format(...)`?

Reply:
No idea, I guess it was longer in the past and I didn't reformat properly :-)